integration_test/test/hls_endpoint_test.exs

```diff
@@ -76,6 +76,52 @@ defmodule Membrane.RTC.HLSEndpointTest do
       check_separate_hls_playlist(output_dir, 2, 2)
     end
 
+    test "creates correct hls stream from single (h264) input, with manual track addition", %{
+      rtc_engine: rtc_engine,
+      tmp_dir: tmp_dir
+    } do
+      file_endpoint_id = "file-endpoint-id"
+
+      file_name = "video.h264"
+      file_path = Path.join(@fixtures_dir, file_name)
+
+      hls_endpoint_id = "hls-endpoint"
+
+      track_id = "test-track-id"
+      stream_id = "test-stream"
+
+      output_dir = Path.join([tmp_dir, stream_id])
+
+      hls_endpoint = create_hls_endpoint(rtc_engine, tmp_dir, :single)
+      hls_endpoint = %{hls_endpoint | subscribe_mode: :manual}
+      :ok = Engine.add_endpoint(rtc_engine, hls_endpoint, id: hls_endpoint_id)
+
+      file_endpoint =
+        create_video_file_endpoint(rtc_engine, file_path, stream_id, file_endpoint_id, track_id)
+
+      :ok = Engine.add_endpoint(rtc_engine, file_endpoint, id: file_endpoint_id)
+
+      assert_receive %Message.EndpointMessage{
+                       endpoint_id: ^file_endpoint_id,
+                       message: :tracks_added
+                     },
+                     @tracks_added_delay
+
+      assert_receive %Message.TrackAdded{track_id: track_id}
+
+      Engine.message_endpoint(rtc_engine, hls_endpoint_id, {:subscribe, [track_id]})
```

Comment:
Before sending a subscribe message, shouldn't we first check that the endpoint doesn't subscribe automatically? Currently, this test passes also for `subscribe_mode` `:auto`, which I assume isn't expected behaviour.

Reply:
we don't have that kind of message in rtc_engine

Reply:
Ok so maybe we should use `refute_receive` or something like that. I think we should construct this test this way that it won't pass if we would change `subscribe_mode: :manual`